### ISSUE TITLE
A2C in-place gradient

### ISSUE THREAD

**Issue by 24neurons** [24neurons/MTRL-Privacy#3]
# A2C buffer backward error
## 1. Name 
`RuntimeError:` Trying to backward through the graph a second time (or directly access saved tensors after they have already been freed). Saved intermediate values of the graph are freed when you call `.backward()` or `autograd.grad()`. Specify `retain_graph=True` if you need to backward through the graph a second time or if you need to access saved tensors after calling backward.
<img width="1061" alt="Screenshot 2024-08-27 at 15 12 39" src="https://github.com/user-attachments/assets/92de401a-ea40-4a89-962a-f32dc8418e87">

https://discuss.pytorch.org/t/inplace-operation-errors-when-implementing-a2c-algorithm/145406/7

## 2. Suspection

This could be raised because of A2C problem performing backpropagation, where we backward through the old buffer variable, where new buffer variable has been replaced.

## 3. Proposed solution 
Probably it is a good practice if we can
      * Erase memory
      * Deep copy
